content/blog/2024-05-22-rustl-2024-unconference.md

```diff
@@ -0,0 +1,196 @@
+
+
+### On the tyranny of struturelessness
+
+*Note: This section is very much my personal opinion and doesn't represent the Linebender project as a whole.*
+
+I'm not sure the self-organized format worked well.
+
+From what I saw, the Embedded team took well to it, and the Rust team was kept productive thanks to Alice Cecile's efforts in marshalling everyone.
+
+In the case of the GUI team, people were spread in a very large room, which should have been conducive to small side-discussions and people splitting up to talk about the problems that interested them. Instead, there was an unspoken accord to progress through agenda items one by one, with a few people dominating the discussions on these items.
+
+Because those people we spread around a large room, they had to talk loudly to address each other, which left little room for side-discussions.
+Because the people talking were naturally the more confident and extraverted, more introverted people ended up taking a passive role in the discussion.
+
+To me, this feels like a strategic mistake.
+It was a setup that encouraged bikeshedding and long back-and-forths, and discouraged plurality of opinions.
+It's no coincidence that the most interesting conversations of the Unconference happened at lunch: lunch was the point of the Unconference where people were most mixed, had the most spontaneous conversations, and were least constrained by having to follow what someone else was saying.
+
+Our discussion were still productive, I just feel like the format could have been improved.
+
+
+## The GUI team
+
+As a member of the Linebender project, I was mostly exposed to the GUI team's proceedings.
+
+The main topics of interest were:
+
+- Build tooling (especially for Android and other non-desktop platforms).
+- Text layout and editing.
+- Winit adoption.
+- Accessibility.
+- Potential for further collaboration.
+- Requests for new Rust features.
+
+Note that I'm skipping a lot topics on the agenda, and other side discussions.
+It feels like those topics were the most prominent during those two days.
+
+There was also a fair amount of informal chatter around LogLog Games' strongly critical [Leaving Rust gamedev after 3 years](https://loglog.games/blog/leaving-rust-gamedev/) article.
+I think those discussions mostly reflected the Bevy community's consensus, that the article had some flaws but overall pointed at very clear weaknesses in the Rust GUI / game engine ecosystem.
+
+### Build tooling
+
+Everyone agreed that a better solution was needed to ship non-TUI Rust apps.
+
+There was discussion about various platforms and build system, and the difficulty of having to eg write Java code for an Android port, and whether anyone wanted to take responsibility for maintaining that glue layer for the rest of the ecosystem.
+
+Unfortunately the discussions on this subject didn't really go anywhere, as far as I could tell.
+Packaging Rust apps is still very much an open problem.
```

Comment:
We've confirmed several times over the years that the Cargo team considers this functionality out of scope for Cargo and does not wish to add it. Note that I agree with this, especially since some packaging tools can only be executed on certain platforms (which defeats the promise of cargo's cross-platform nature), so _true_ multi-target builds that cover all platforms of interest aren't even possible.